functions/logStripeEvent.js

```diff
@@ -7,22 +7,51 @@ if (admin.apps.length === 0) {
 
 const DEFAULTCPL = 1.0;
 
+// create a dummy payment_intent object and send it through the system
+exports.testPaymentIntent = functions.https.onRequest(async (req, res) => {
+  const event = {
+    id: 'fake-event-id',
+    data: {
+      object: {
+        description: 'Give Lively / Smart Donations',
+        amount: 2076,
+        metadata: {
+          transaction_fee_covered_by_donor: '$0.76',
+          user_email: 'fake@email.biz',
+          user_first_name: 'fakeName',
+          utm_source: 'Africa|Africa|fakeReferral',
+        },
+      },
+    },
+  };
+  console.log('testing donation pathway');
+  const msg = this.handlePaymentIntentSucceeded(event.data.object, event.id);
+  return res.status(200).send({msg: msg, obj: event});
+});
+
 exports.logPaymentIntent = functions.https.onRequest(async (req, res) => {
   const event = req.body;
   if (!req.body) {
     return res.status(400).send('no data supplied!');
   }
   console.log(`parsing event with id ${event.id}`);
+  let intent;
+  let msg;
   switch (event.type) {
     case 'payment_intent.succeeded':
-      const paymentIntent = event.data.object;
+      intent = event.data.object;
       if (paymentIntent.description === 'Give Lively / Smart Donations') {
-        console.log(`successful payment for ${paymentIntent.amount}`);
-        this.handlePaymentIntentSucceeded(paymentIntent, event.id);
+        console.log(`successful payment for ${intent.amount}`);
+        msg = this.handlePaymentIntentSucceeded(intent, event.id);
       }
       break;
   }
-  return res.status(200).send({msg: 'sucessfully received event', data: event});
+  if (msg.data.err) { // check to see if the data were successfully parsed
+    res.status(501);
```

Comment:
This should be a 500 error.  501 errors are for something that has not been implemented yet.  Like if you hit a `/v3` route instead of a `/v2` route for API versioning.

Reply:
oh! thank you for the clarification